psl-core/src/main/java/org/linqs/psl/application/inference/online/OnlineInference.java

```diff
@@ -91,6 +108,46 @@ protected void executeAction(OnlineMessage action) {
         server.onActionExecution(action, new ActionStatus(action, true, response));
     }
 
+    protected String doAddAtom(AddAtom action) {
+        GroundAtom atom = null;
+
+        if (atomManager.getDatabase().hasAtom(action.getPredicate(), action.getArguments())) {
+            atom = ((OnlineAtomManager)atomManager).deleteAtom(action.getPredicate(), action.getArguments());
+            ((OnlineTermStore)termStore).deleteLocalVariable(atom);
+        }
+
+        if (action.getPartitionName().equalsIgnoreCase("READ")) {
+            atom = ((OnlineAtomManager)atomManager).addObservedAtom(action.getPredicate(), action.getValue(), action.getArguments());
+        } else {
+            atom = ((OnlineAtomManager)atomManager).addRandomVariableAtom(action.getPredicate(), action.getValue(), action.getArguments());
+        }
+
+        atom = ((OnlineTermStore)termStore).createLocalVariable(atom);
```

Comment:
Leave out the assignment here.
I want to avoid people thinking that `atom` may be changed suring `createLocalVariable`.